### PR TITLE
feat: x/prices slinky adapter

### PR DESCRIPTION
**Pull request by Eric-Warehime**
Methods required for the x/prices module to be compatible with the [currency pair](https://github.com/skip-mev/slinky/blob/main/abci/strategies/currencypair/types.go#L14) OracleKeeper interface.

BLO-826

### CODE REVIEW

File: protocol/testutil/keeper/prices.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ func CreateNMarkets(t *testing.T, ctx sdk.Context, keeper *keeper.Keeper, n int)
 	numExistingMarkets := GetNumMarkets(t, ctx, keeper)
 	for i := range items {
 		items[i].Param.Id = uint32(i) + numExistingMarkets
-		items[i].Param.Pair = fmt.Sprintf("%v", i)
+		items[i].Param.Pair = fmt.Sprintf("%v-%v", i, i)
```
Comment (g3ndo9kar9) [inline protocol/testutil/keeper/prices.go:126]: why do we do this?

File: protocol/x/prices/keeper/slinky_adapter.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package keeper
+
```
Comment (g3ndo9kar9) [inline protocol/x/prices/keeper/slinky_adapter.go:2]: nit: let's move this to a separate dir, /adapters?

File: protocol/x/prices/keeper/slinky_adapter.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+func MarketPairToCurrencyPair(marketPair string) (oracletypes.CurrencyPair, error) {
+	split := strings.Split(marketPair, "-")
+	if len(split) != 2 {
+		return oracletypes.CurrencyPair{}, fmt.Errorf("incorrectly formatted CurrencyPair: %s", marketPair)
+	}
+	cp := oracletypes.CurrencyPair{
+		Base:  strings.ToUpper(split[0]),
+		Quote: strings.ToUpper(split[1]),
```
Comment (g3ndo9kar9) [inline protocol/x/prices/keeper/slinky_adapter.go:68]: We have to account for Decimal here?

**Review by g3ndo9kar9 — APPROVED** [review]
lgtm, assuming this isn't using the latest decimals changes from slinky

File: protocol/x/prices/keeper/slinky_adapter.go (hunk 2 above)
Comment (Eric-Warehime) [reply protocol/x/prices/keeper/slinky_adapter.go:2]: We can. There's only one there for now--not sure if there will be more ever?

File: protocol/testutil/keeper/prices.go (hunk 1 above)
Comment (Eric-Warehime) [reply protocol/testutil/keeper/prices.go:126]: The assumption I'm making here is that pairs in the dydx market params are formated `BASE-QUOTE` based on https://github.com/skip-mev/v4-chain/blob/main/protocol/x/prices/testdata/expected_default_genesis.json

Whereas our currency pair strings are `BASE/QUOTE`. 

The keeper adapter will just ignore (and log) anything which doesn't conform to this convention.

File: protocol/x/prices/keeper/slinky_adapter.go (hunk 3 above)
Comment (Eric-Warehime) [reply protocol/x/prices/keeper/slinky_adapter.go:68]: I haven't done any decimal/exponent stuff in this--keeping it to just fetching/storing prices. We may or may not need to do conversions somewhere in here.

**Pull request merged**